### PR TITLE
refactor(website): declare every theme colour once via light-dark()

### LINKED ISSUE #1216: refactor(website): adopt light-dark() and fix scaffold-guideline drift
## Problem

The marketing site (`website/`) is the framework's flagship dogfood app, but it does not follow the styling and layout guidance WebJs ships to its own users. Three concrete divergences, all verified against the current tree at `4822e0d1`:

**1. The palettes duplicate every dark value instead of using `light-dark()`.**

`.agents/skills/webjs/references/styling.md` (the "Light and dark, defined once (DRY)" section, L112-142) and `packages/cli/lib/create.js` (L1245-1330, the generated `app/layout.ts` token block) both teach one rule: write each colour token ONCE with the native `light-dark(LIGHT, DARK)` function and let `color-scheme` pick the side. The website does the opposite, in two places:

- `website/app/layout.ts` L293-323: the dark palette is written TWICE, once under `@media (prefers-color-scheme: dark) { :root:not([data-theme='light']) }` and again verbatim under `:root[data-theme='dark']`. 16 tokens, ~30 lines of pure duplication. The two blocks are already byte-equal, so any future edit has to be made in both or the OS-dark and toggle-dark paths silently diverge.
- `website/public/input.css` L152-218: the `.ui-preview` component-preview palette writes 19 tokens THREE times (base, `@media (prefers-color-scheme: dark)`, `:root[data-theme='dark']`), ~66 lines.

The prerequisite for `light-dark()` is already in place: `website/app/layout.ts` L248 declares `color-scheme: light dark` on `:root`, L324 sets `:root[data-theme='light'] { color-scheme: light }` and L310 sets `:root[data-theme='dark'] { color-scheme: dark }`. So the three-way selector logic is redundant with a `color-scheme` cascade that already exists.

**2. `website/components/ui/` holds 11 accidentally-committed dead files.**

`website/AGENTS.md` states that `components/ui/` is "intentionally EMPTY here, left free for `webjs ui add` to own, exactly as the scaffold expects". It is not empty: commit `235d0fb0` (#1213) added 11 files / 766 lines there. Every one is BYTE-IDENTICAL to its counterpart in the gitignored generated mirror `website/modules/ui/components/` (verified with `diff` on `card.ts`, `kbd.ts`, `table.ts`), and NONE is imported by anything in the app (`grep` for each basename across `app lib modules components` returns zero non-self hits).

They leaked because `website/.gitignore` ignores `/modules/ui/components/`, `/lib/utils/cn.ts`, and `/lib/utils/dom.ts` but NOT `/components/ui/`.

**3. Doc drift in `website/AGENTS.md`.**

The "Announcement banner" section describes a strip the root layout renders "just above the sticky header". No such markup exists in `app/layout.ts` any more, and L329 explicitly comments `--header-h: 59px; /* #610 fixed header offset (no banner) */`. The same section calls the header "sticky" when #610 deliberately made it `position: fixed`.

## Design / approach

**1. Collapse both palettes onto `light-dark()`.**

For `app/layout.ts`, fold each duplicated dark colour back into its `:root` declaration as `light-dark(<light>, <dark>)` and delete both dark blocks, keeping only the `color-scheme` switches. Per the styling reference's "Edge cases" paragraph, `light-dark()` is COLOUR-only, so the non-colour and composite tokens must keep an explicit override pair:

- `--glow-strength` (`0.16` / `0.08`) is a bare number, not a colour.
- `--shadow-sm` / `--shadow` are full shadow values (geometry + colour), so `light-dark()` cannot wrap them whole. Either keep the override pair or split the colour out into its own `light-dark()` token and build the shadow from it (preferred: it puts the per-theme part back under the single-definition rule).
- `--accent-tint`, `--cta-surface`, `--accent-surface`, `--accent-border` are DERIVED via `color-mix()` from `--accent-live`; once `--accent-live` is a `light-dark()` token they track both themes for free and their overrides can go.

For `public/input.css`, the `.ui-preview` block is a scoped container rather than `:root`, which is fine: `color-scheme` inherits, so `.ui-preview` picks up whatever `:root` resolved and `light-dark()` inside it selects the matching side with no extra selectors.

**2. Delete `website/components/ui/*.ts` and add `/components/ui/` to `website/.gitignore`,** restoring the state `website/AGENTS.md` already documents and keeping the directory free for `webjs ui add`.

**3. Correct the two stale paragraphs in `website/AGENTS.md`.**

## Implementation notes (for the implementing agent)

**Where to edit**

- `website/app/layout.ts` L247-324: the `:root` token block, the `@media (prefers-color-scheme: dark)` block, the `:root[data-theme='dark']` block, and the two `color-scheme` switches. This is inside a template literal in a page/layout, which is legal (a layout never hydrates, so interpolating into `<style>` is allowed; invariant 9's backtick ban still applies inside the `html` tag).
- `website/public/input.css` L152-218: the `.ui-preview` palette and its two dark blocks. Do NOT touch the `@custom-variant dark` block at L23-28 or the `@theme` / `@theme inline` blocks at L30-137; the custom variant is already correctly keyed off `[data-theme]` rather than the kit's `.dark`, which is what keeps the previews in step with the page theme.
- `website/components/ui/*.ts`: delete all 11 files.
- `website/.gitignore`: add `/components/ui/`.
- `website/AGENTS.md`: the "Announcement banner" section and the `components/` layout block.

**Landmines / gotchas**

- `website/scripts/copy-registry.mjs` L44-63 carries a one-time LEGACY migration that DELETES files from `components/ui/` whose contents match `/from '(\.\/lib\/|\.\.\/\.\.\/lib\/ui\/)/`. The 11 committed files have NO imports at all, so they do not match and survive every dev cycle. Deleting them by hand is therefore required; the script will not do it. Leave that migration block alone.
- `website/public/tailwind.css` is a GITIGNORED compiled artifact. Edit `public/input.css` only, then let `webjs.dev.before` / `webjs.dev.regenerate` recompile (see `website/AGENTS.md` "Run"). Never hand-edit `tailwind.css`.
- The `.ui-preview` palette carries a comment (L144-150) saying it mirrors what `webjs ui init` writes into a user's `globals.css`, i.e. `packages/ui/packages/registry/themes/index.css`. That file is NOT actually mirrored verbatim today: it uses `@custom-variant dark (&:is(.dark *))` plus `:root` / `.dark` blocks (L55, L66, L102), while the website keys off `[data-theme]`. If `.ui-preview` moves to `light-dark()` and the registry theme does not, update that comment so it stops claiming a byte-for-byte mirror it never was. Converting the registry theme itself is a SCAFFOLD surface change and belongs in a separate issue (it would need the `webjs-scaffold-sync` skill and a generate-boot-check pass).
- Verify in a REAL browser in all three toggle states (`system` with a dark OS, forced `light`, forced `dark`) plus the `/ui` gallery previews. Per the styling reference: "Light mode passing proves nothing about dark." The `<theme-toggle>` at `website/components/theme-toggle.ts` writes `data-theme` on `<html>`; the no-FOUC bootstrap that reads it back is the inline script at `app/layout.ts` L128-134.
- `light-dark()` is CSS Color 5 / Baseline 2024. All four current targets (the site's supported browsers) have it, but `oklch()` values inside it are fine since `light-dark()` accepts any `<color>`.

**Invariants to respect**

- AGENTS.md invariant 7 (light-DOM tag-prefix) does not apply to the layout's `<style>`: those rules belong to a layout, not a component. Do not "fix" them.
- AGENTS.md invariant 11 (prose punctuation and `WebJs` brand casing) applies to every markdown line touched in `website/AGENTS.md`.
- The root layout is the ONE file allowed to write its own shell (invariant 8). Do not move the token block anywhere else.

**Tests + docs surfaces**

- `website/test/ssr/*` is the existing SSR suite; `website/test/ssr/ui-gallery.test.ts` already pins `seed: false`. Add an SSR assertion that the rendered root layout emits `light-dark(` and NO `@media (prefers-color-scheme: dark)` palette block, with the counterfactual (revert the layout, the test reds).
- Add a test that `website/components/ui/` stays empty, so the accidental re-commit cannot recur silently.
- Run `npx webjs check` in `website/` (currently clean) and `npm test`.
- No framework `packages/*/src` change here, so the doc gate does not apply; `website/AGENTS.md` is the doc surface for this one.

## Acceptance criteria

- [ ] `website/app/layout.ts` declares every per-theme COLOUR token once via `light-dark()`; the `@media (prefers-color-scheme: dark)` and `:root[data-theme='dark']` palette duplicates are gone, leaving only `color-scheme` switches plus the documented non-colour overrides
- [ ] `website/public/input.css` `.ui-preview` declares its 19 tokens once via `light-dark()`; its two dark blocks are gone
- [ ] Non-colour per-theme tokens (`--glow-strength`, the shadows) are handled per the styling reference's edge-case rule, not silently dropped
- [ ] `website/components/ui/` is empty and `/components/ui/` is gitignored
- [ ] `website/AGENTS.md` no longer documents a non-existent announcement banner, and describes the header as fixed (#610) rather than sticky
- [ ] Rendered colours are unchanged in all three theme states, verified in a real browser on the home page, a `/docs` page, and a `/ui` component page
- [ ] A counterfactual proves the new SSR test actually fires
- [ ] `npx webjs check` and `npm test` pass in `website/`

### PR DESCRIPTION
**Pull request by vivek7405**
Closes #1216

Brings the marketing site in line with the styling and layout rules WebJs ships to its own users. It was the counter-example to its own guidance on three counts.

## Each theme colour is declared once, via `light-dark()`

`references/styling.md` in the skill, and the palette `webjs create` generates, both teach one rule: write each colour token ONCE as `light-dark(LIGHT, DARK)` and let `color-scheme` pick the side. The site did the opposite everywhere it had a palette.

Four places wrote it twice instead, verbatim, once under `@media (prefers-color-scheme: dark)` and again under `:root[data-theme='dark']`:

- `app/layout.ts`, the site palette (16 colours).
- `public/input.css`, the `.ui-preview` kit palette (19 colours, said three times over).
- `public/input.css`, the syntax-highlight classes (5 colours).
- `app/page.ts`, the home page's code-sample tokens (3 colours).

Every pair was already byte-equal, so any future edit had to land in both copies or the OS-dark and toggle-dark paths would drift apart silently.

The prerequisite was already there: `:root` declares `color-scheme: light dark` and the toggle's two attribute rules force one. So the three-way selector logic was redundant with a cascade that already existed.

Three tokens genuinely cannot ride `light-dark()`, which is colour-only, so they keep the explicit override pair the styling reference prescribes:

- `--glow-strength` is a bare number.
- `--cta-mix` is the CTA fill's mix ratio, split out of `--cta-surface` because the two themes differ by the ratio, not by a colour.
- `--shadow-spread` is the geometry half of `--shadow`, whose cast widens in dark mode. Its two colours became their own `light-dark()` tokens, and `--shadow-sm` keeps identical geometry in both themes so its colour rides `light-dark()` inline.

Everything derived through `color-mix()` (`--accent-tint`, `--accent-surface`, `--accent-border`, `--shadow-glow`) now tracks both themes with no override at all.

## `components/ui/` is empty again

`website/AGENTS.md` says the directory is intentionally empty, left free for `webjs ui add` to own exactly as in a scaffolded app, with the gallery's preview copies living in the generated `modules/ui/components/` mirror. It was not empty: eleven files (766 lines) landed in it in #1213, every one byte-identical to its counterpart in that mirror and imported by nothing. `.gitignore` covered the other two generated paths but not this one, which is how they got in. Now ignored as well as removed.

## Doc drift

`website/AGENTS.md` documented an announcement banner above a "sticky" header. The banner is gone and #610 made the header `position: fixed` deliberately, because a sticky one flickers on iOS WebKit during a client-router nav. Also typed the search debounce handle, which was `any`.

## Test plan

- **Colour equivalence, in a real browser.** 360 resolved readings, every palette token plus the highlight classes and code tokens, in all six theme states (OS light and dark, each with the toggle unset, forced light, forced dark), compared against a pristine `main` build on a second port. Zero differences. Shadows are read back through `box-shadow` rather than as custom properties, since a custom property's computed value keeps `light-dark()` unsubstituted, so reading the property compares source text rather than what the browser paints. That distinction was load-bearing: the first probe pass compared raw strings and reported eighteen false differences.
- **Unit.** `test/ssr/design-tokens.test.ts` rewritten to parse the `light-dark()` pairs (its old parser read the now-deleted dark block) and extended to pin the DRY rule itself: only the three non-colour tokens may carry a per-theme override, and all three `color-scheme` declarations must be present. New `test/ssr/kit-surfaces.test.ts` pins the empty `components/ui/`, the `.ui-preview` palette's shape, and sweeps every stylesheet, page, layout, and component asserting the rule itself: no colour under a per-theme selector, only the three non-colour tokens. That sweep replaced a per-file assertion, which is what let the highlight classes and code tokens survive the first pass.
- **Counterfactuals**, all eight proven to fire at [`7642b099`](https://github.com/webjsdev/webjs/commit/7642b099): reintroducing a colour override under a theme selector, reverting one token to a bare declaration, dropping the forced-dark `color-scheme`, committing a file into `components/ui/`, re-adding a duplicated `.ui-preview` dark block, and the three shapes the sweep catches (a bare colour on a class under a dark selector, a custom-property colour under a media query, a colour token in a page's own style block). Each reds the suite; restoring greens it.
- **Suites.** 165 server tests and 35 browser tests pass. `webjs check` and `webjs typecheck` clean.
- **Boot.** Serves 200 on `/`, `/docs/getting-started`, `/ui`, `/ui/button`, `/brand` in prod mode.

## Doc surfaces

- `website/AGENTS.md`: updated (the styling rule, the header section, the `components/ui/` note).
- The skill at `.agents/skills/webjs/` and `packages/cli/lib/create.js`: N/A, both already teach the `light-dark()` rule. This PR makes the site follow it, not the other way round.
- Docs site, marketing copy, MCP, editor plugins, scaffold templates, framework `packages/*/src`: N/A, no framework surface changed.
- Bun parity: N/A, no runtime-sensitive source touched.

## Deliberately not done here

The `.ui-preview` comment claimed to mirror `packages/ui/packages/registry/themes/index.css` byte for byte. It never did (the registry keys its dark half off a `.dark` class, this site off `[data-theme]`), so the comment is corrected to say which parts are shared and which differ. Converting the registry theme itself to `light-dark()` is a scaffold surface change, needs the `webjs-scaffold-sync` skill and a generate-boot-check pass, and belongs in its own issue.

### CODE REVIEW

**Comment by vivek7405** [comment]
**Design rationale: why the shadow tokens split, and why the check became a sweep**

Two calls worth writing down.

**The shadow split.** `light-dark()` is colour-only, so a shadow cannot ride it whole. I had three options: leave both shadows in the override block, split their colours out into `light-dark()` tokens, or reconstruct the whole shadow from geometry plus colour tokens. I went with the middle one, unevenly, because the two shadows are not the same case. `--shadow-sm` has identical geometry in both themes, so its colour is the ONLY per-theme part and it rides `light-dark()` inline with nothing extra. `--shadow` genuinely changes spread (`0 8px 30px` light, `0 10px 40px` dark, because a tighter cast disappears over black), so it needs a `--shadow-spread` token regardless. Splitting its two colours out anyway means the override block holds only the part that truly cannot be expressed, which is the line the styling reference draws.

`--cta-mix` is the same call in a smaller place. The CTA fill's two themes differ by the mix RATIO (7% against 6%), not by a colour, so hoisting the ratio out lets `--cta-surface` stay a single definition instead of being restated per theme for the sake of one digit.

**Why the check is a sweep.** My first version asserted the two palettes I had found by hand. That is exactly the check that would not have found the other two: the syntax-highlight classes in `input.css` and the home page's code tokens were both duplicated the same way, and I only caught them by grepping for theme selectors afterwards rather than by any test. So the assertion is now the rule itself, swept over every stylesheet and every page, layout, and component: no colour under a per-theme selector, only the three non-colour tokens. Docs `pre` blocks are excluded, since a code sample teaching the reader is not this site's styling.

Two things the sweep taught me while writing it, both of which cost a counterfactual to find. Comments have to be stripped before matching, or prose that NAMES a selector to explain why the block does not use it reads as a live rule. And the block terminator cannot anchor to a newline: `:root[data-theme='dark'] .t-str { color: ... }` is a one-liner, which is precisely the shape the highlight classes used, so the first version walked straight past the case it was written for.

File: website/lib/env.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * server-only (two metadata routes, a route handler, and a `.server.ts`), so
+ * `process` does exist for all of them. It is written this way so the module
+ * stays importable from a browser-loading module too, since the origin is
+ * public information already present in the rendered HTML and there is nothing
+ * here that needs the server boundary. A bare `process.env` would throw at
+ * module load the moment such an importer appeared.
+ */
+type SiteEnv = { SITE_URL?: string };
+
+/**
```
Comment (vivek7405) [inline website/lib/env.ts:20]: `env()` is exported but nothing outside this module calls it. `siteUrl()` is the only consumer, and the `NODE_ENV` member of `SiteEnv` is never read either. This commit removes a dead env read from `lib/links.ts` and adds a dead export here, which is a bad trade.

File: website/lib/env.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * The site's canonical origin, read once from the environment.
+ *
+ * Four modules (app/sitemap.ts, app/robots.ts, app/llms.txt/route.ts, and
+ * lib/docs-llms.server.ts) each wrote their own
+ * `(globalThis as any).process?.env?.SITE_URL` for this, which is one missing
```
Comment (vivek7405) [inline website/lib/env.ts:6]: This names `lib/links.ts` as a browser-loading consumer to justify the `globalThis` guard, but the same commit deletes that consumer. All four remaining callers are server-only (two metadata routes, a route handler, a `.server.ts`), so the stated reason is false as of this diff. The guard is still worth keeping, just not for this reason.

**Review by vivek7405 — COMMENTED** [review]
Went back over this one properly. The palette work is the strong part: collapsing four duplicated dark blocks onto `light-dark()` is a clear net negative in lines, the three non-colour tokens that genuinely cannot ride it are carved out for the right reason rather than left behind by accident, and the 360-reading browser comparison against main is the kind of evidence I want on a refactor that touches every colour on the site. The sweep test is the piece I would keep longest: asserting the rule instead of the two known sites is what caught the highlight classes and the code tokens, which a per-file check had already walked past once.

Two things I got wrong, both in `lib/env.ts`, both mine from this branch.

I also checked two things I had half-suspected and cleared both. The `ref()` binding leaves no artifact in the SSR output (`<copy-cmd data-wj-host><!--webjs-hydrate-->`), so the byte-stability the conditional-GET path depends on is intact. And `updateComplete.then()` with no `.catch()` is safe here: `component.js` is explicit that the promise always resolves and never rejects, so there is no unhandled rejection to guard against.

Fixed in 6664b07f.

File: website/lib/env.ts (hunk 1 above)
Comment (vivek7405) [reply website/lib/env.ts:20]: Fixed in [`6664b07f`](https://github.com/webjsdev/webjs/commit/6664b07f).

File: website/lib/env.ts (hunk 2 above)
Comment (vivek7405) [reply website/lib/env.ts:6]: Fixed in [`6664b07f`](https://github.com/webjsdev/webjs/commit/6664b07f).
